components/compass-runtime-agent/internal/compassconnection/supervisor.go

```diff
@@ -121,6 +124,15 @@ func (s *crSupervisor) SynchronizeWithCompass(connection *v1alpha1.CompassConnec
 		return s.updateCompassConnection(connection)
 	}
 
+	// TODO(mszostok): bug with certs input - outdated?
```

Comment:
we need to discuss if there is a bug with outdated `credentials`.  In line 120 there is a method which can refresh the credentials but it does not return the new one if refreshed.

Reply:
If credentials are refreshed, they are saved to the Secret. We could return them, but current credentials still need to be valid to be refreshed so unless credentials expired just after a call to refresh and before the call to Director, there should be no issue.